headspring/headspring.py

```diff
@@ -0,0 +1,115 @@
+import base64
+import logging
+import httplib2
+import hashlib
+import json
+import os
+import sys
+import traceback
+import uuid
+
+from oauth2client.client import GoogleCredentials
+from apiclient import discovery
+from flask import Flask, abort, jsonify, request
+
+from tools import generate_id
+from ConfigParser import ConfigParser
+
+
+os.chdir(os.path.dirname(sys.argv[0]))  # change into execution directory
+config = ConfigParser()
+
+try:
+    with open('config') as _fp:
+        config.readfp(_fp)
+except Exception as e:
+    logging.info('failed to open config file')
+
+
+PROJ_NAME = config.get('override', 'proj_name')
+
+app = Flask(__name__)
+def get_pubsub_client(version='v1beta2'):
```

Comment:
this function exists in both this file and tools, seems like its a good fit for tools so we can share it between producers easier.